### ISSUE TITLE
Invoke-CredentialInjection does not work on Windows 10

### ISSUE THREAD

**Issue by okazymyrov** [PowerShellMafia/PowerSploit#151]
I have tried Invoke-CredentialInjection with different roles (user, Administrator, SYSTEM) on Windows 10 x64 without success. The error is

``` powershell
Invoke-MemoryLoadLibrary : Specified cast is not valid.
At line:2807 char:24
+ ... oadedInfo = Invoke-MemoryLoadLibrary -PEBytes $PEBytes -ExeArgs $ExeA ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (:) [Invoke-MemoryLoadLibrary], InvalidCastException
    + FullyQualifiedErrorId : System.InvalidCastException,Invoke-MemoryLoadLibrary
```

**Comment by HarmJ0y** [comment]
Thanks for the information. We will be doing a refactor of a chunk of PowerSploit code at some point after BlackHat and we will dive into this cast issue then.

**Comment by FuzzySecurity** [comment]
This issue has bothered me for a long time so I put on some coffee and traced through, it seems like an explicit cast was missing in Invoke-MemoryLoadLibrary. See my commit on the Dev branch:
https://github.com/PowerShellMafia/PowerSploit/commit/b895866c3b5d6a879180ae3db3019de5c258e050

The PowerShell error was really very unhelpful in this case!

While testing the fix I also found another issue (Win7), I expect this is a PSv2 problem:
![image](https://user-images.githubusercontent.com/10997340/28288406-f9e7845a-6b36-11e7-9f0a-9f03c46b31cf.png)

I pushed a fix for that as well:
https://github.com/PowerShellMafia/PowerSploit/commit/e47c52a090b6b590e6632eecf705bf566f3c74af

Closing this issue now!
